lib/fakezoomapi.ts

```diff
@@ -1,10 +1,45 @@
-import { DrawImageCallback, ZoomApiWrapper } from "./zoomapi";
+import {
+  DrawImageCallback,
+  ZoomApiWrapper,
+  AuthorizeCallback,
+  AuthorizeOptions,
+  GeneralMessageResponse,
+} from "./zoomapi";
 
 class FakeZoomApi implements ZoomApiWrapper {
+  private authorizeCallback: AuthorizeCallback | null = null;
+  constructor(private inCypress: boolean) {}
+
   async setDrawImageCallback(cb: DrawImageCallback): Promise<void> {}
+  async authorize(data: AuthorizeOptions): Promise<GeneralMessageResponse> {
+    return { message: "Success" };
+  }
+  async setAuthorizeCallback(cb: AuthorizeCallback): Promise<void> {
+    this.authorizeCallback = cb;
+    if (this.inCypress) {
+      setTimeout(() => cb({ code: "mocked_code" }), 100);
+    }
+  }
+
+  async triggerAuthorizeCallback(event: { code: string }) {
+    if (this.authorizeCallback) {
+      await this.authorizeCallback(event);
+    }
+  }
 }
 export type { ZoomApiWrapper };
 
-export function createFromConfig() {
-  return new FakeZoomApi();
+function createFakeZoomApi(inCypress: boolean = false) {
+  console.log("creating fake zoom api:", inCypress);
+  return new FakeZoomApi(inCypress);
+}
+
+let zoomApi: ZoomApiWrapper;
+
+if (typeof window !== "undefined" && window.Cypress) {
+  zoomApi = createFakeZoomApi(true);
+} else {
+  zoomApi = createFakeZoomApi(false);
 }
```

Comment:
Can we revert this part now that we have the `NEXT_PUBLIC_MOCK_ZOOM_API`?

Ideally, I'd like the behavior when setting `NEXT_PUBLIC_MOCK_ZOOM_API` to be the same in Cypress as in a normal browser and not have any special casing for Cypress anywhere